<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Test Custom Contrasts with LIMMA — limma_contrasts • MSnSet.utils</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Test Custom Contrasts with LIMMA — limma_contrasts"><meta property="og:description" content="A convenience wrapper for differential analysis with LIMMA using custom
contrasts."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MSnSet.utils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/complex_heatmap.html">Create Annotated Expression and Correlation Heatmaps</a>
    </li>
    <li>
      <a href="../articles/fuzzy_cmeans.html">Fuzzy C-Means</a>
    </li>
    <li>
      <a href="../articles/heatmap_examples.html">Heatmaps</a>
    </li>
    <li>
      <a href="../articles/heatmap.plus.2.html">heatmap.plus.2</a>
    </li>
    <li>
      <a href="../articles/id_conversion.html">ID Conversion</a>
    </li>
    <li>
      <a href="../articles/ontology_overrepresentation.html">Over-representation Analysis</a>
    </li>
    <li>
      <a href="../articles/PeCorA.html">PeCorA</a>
    </li>
    <li>
      <a href="../articles/peptide_mapping_example_1.html">Peptide Mapping Example 1</a>
    </li>
    <li>
      <a href="../articles/plot_sel_feat_with_p_vals.html">Example of MSnSet Feature Bar Chart with P-Value Coloring</a>
    </li>
    <li>
      <a href="../articles/wgcna_example.html">Example of WGCNA Analysis</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Test Custom Contrasts with LIMMA</h1>

    <div class="hidden name"><code>limma_contrasts.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A convenience wrapper for differential analysis with LIMMA using custom
contrasts.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">limma_contrasts</span><span class="op">(</span></span>
<span>  <span class="va">eset</span>,</span>
<span>  <span class="va">model.str</span>,</span>
<span>  <span class="va">coef.str</span>,</span>
<span>  <span class="va">contrasts</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"t"</span>, <span class="st">"F"</span><span class="op">)</span>,</span>
<span>  var.group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  col.group <span class="op">=</span> <span class="va">coef.str</span>,</span>
<span>  adjust.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  remove_name <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-eset">eset<a class="anchor" aria-label="anchor" href="#arg-eset"></a></dt>
<dd><p><code>eSet</code> (or most likely <code>eSet</code> subclass) object</p></dd>


<dt id="arg-model-str">model.str<a class="anchor" aria-label="anchor" href="#arg-model-str"></a></dt>
<dd><p>character; formulation of the model (e.g. <code>"~ 0 + a +
  b"</code>). This should be a no-intercept model (it should include 0 or -1 as
terms). See <code><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></code> for more details.</p></dd>


<dt id="arg-coef-str">coef.str<a class="anchor" aria-label="anchor" href="#arg-coef-str"></a></dt>
<dd><p>character; coefficient of interest. One of
<code>colnames(pData(eset))</code>. E.g. <code>"a"</code>.</p></dd>


<dt id="arg-contrasts">contrasts<a class="anchor" aria-label="anchor" href="#arg-contrasts"></a></dt>
<dd><p>character; contrasts to test. All factor levels in the
contrasts should begin with the name of that factor. It is recommended to
use <code><a href="paircomp.html">paircomp</a></code> with the <code>name</code> argument specified to
create pairwise contrasts of the form <code>"a-b"</code>.</p></dd>


<dt id="arg-stat">stat<a class="anchor" aria-label="anchor" href="#arg-stat"></a></dt>
<dd><p>character; the moderated test statistic to use. Either <code>"t"</code>
or <code>"F"</code>.</p></dd>


<dt id="arg-var-group">var.group<a class="anchor" aria-label="anchor" href="#arg-var-group"></a></dt>
<dd><p><code>NULL</code> or character; the column in <code>pData(eset)</code>
indicating groups that will have different relative quality weights. The
default (<code>NULL</code>), means that each sample will be weighted equally.</p></dd>


<dt id="arg-plot">plot<a class="anchor" aria-label="anchor" href="#arg-plot"></a></dt>
<dd><p>logical; whether to generate diagnostic plots. The default
(<code>TRUE</code>) generates a barplot of weights applied to each sample and a
plot of the mean-variance trend using <code><a href="https://rdrr.io/pkg/limma/man/plotSA.html" class="external-link">plotSA</a></code>.</p></dd>


<dt id="arg-col-group">col.group<a class="anchor" aria-label="anchor" href="#arg-col-group"></a></dt>
<dd><p>character; the column in <code>pData(eset)</code> used to color
the text of the MDS plot.</p></dd>


<dt id="arg-adjust-method">adjust.method<a class="anchor" aria-label="anchor" href="#arg-adjust-method"></a></dt>
<dd><p>method for p-value adjustment. Default is <code>"BH"</code>
(Benjamini-Hochberg). See <code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></code> for details.</p></dd>


<dt id="arg-remove-name">remove_name<a class="anchor" aria-label="anchor" href="#arg-remove-name"></a></dt>
<dd><p>if <code>TRUE</code> (the default) <code>coef.str</code> will be
removed from the contrasts. This simply improves readability, but it will
not work properly if <code>coef.str</code> is a substring of any group name.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional arguments passed to <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>data.frame</code>. Basically the output of
<code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></code> with additional columns <code>feature</code> and
<code>contrast</code> (if <code>stat = "t"</code>). All columns from <code>fData</code> are
also included.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>An MDS plot is used to determine the appropriate value of
<code>var.group</code>. If samples within phenotype groups tend to cluster well
and have similar dispersions, the default <code>var.group = NULL</code> is
recommended. If samples within phenotype groups tend to cluster well, but
different groups are more or less dispersed, it may be beneficial to set
<code>var.group</code> to the name of the phenotype column. If one or more
samples tend to cluster poorly with samples of the same phenotype, it may
be beneficial to weight by sample. That is, set <code>var.group</code> to the
name of the column in <code>pData(eset)</code> that uniquely identifies each
sample. If variation between samples or groups of samples is biological
rather than technical, weighting is not recommended.</p>
<p>The plot of the mean-variance trend helps determine whether to fit a trend
line to the prior variance (default is <code>trend = FALSE</code>). It also helps
determine if the prior variance should be robustified against outlier
sample variances (<code>robust = TRUE</code>; default is <code>FALSE</code>). See
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></code> for more details.</p>
<p>p-values are adjusted across all contrasts (globally). It is recommended to
adjust p-values from similar contrasts together. See the "Multiple Testing
Across Contrasts" section of the LIMMA User's Guide (linked below) for more
information.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Liu, R., Holik, A. Z., Su, S., Jansz, N., Chen, K., Leong, H. S.,
Blewitt, M. E., Asselin-Labat, M. L., Smyth, G. K., &amp; Ritchie, M. E.
(2015). Why weight? Modelling sample and observational level variability
improves power in RNA-seq analyses. <em>Nucleic acids research</em>, <em>43</em>(15),
e97. <a href="https://doi.org/10.1093/nar/gkv412" class="external-link">https://doi.org/10.1093/nar/gkv412</a></p>
<p>Phipson, B., Lee, S., Majewski, I. J., Alexander, W. S., &amp; Smyth, G. K.
(2016). ROBUST HYPERPARAMETER ESTIMATION PROTECTS AGAINST HYPERVARIABLE
GENES AND IMPROVES POWER TO DETECT DIFFERENTIAL EXPRESSION. <em>The annals of
applied statistics</em>, <em>10</em>(2), 946–963.
<a href="https://doi.org/10.1214/16-AOAS920" class="external-link">https://doi.org/10.1214/16-AOAS920</a></p>
<p>Smyth G. K. (2004). Linear models and empirical bayes methods for assessing
differential expression in microarray experiments. <em>Statistical
applications in genetics and molecular biology</em>, <em>3</em>, Article3.
<a href="https://doi.org/10.2202/1544-6115.1027" class="external-link">https://doi.org/10.2202/1544-6115.1027</a></p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf" class="external-link">LIMMA
User's Guide</a>, <code><a href="limma_a_b.html">limma_a_b</a></code>, <code><a href="limma_a_b.html">limma_gen</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># library(MSnSet.utils)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cptac_oca</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># A no-intercept model is required. Testing SUBTYPE contrasts</span></span></span>
<span class="r-in"><span><span class="co"># with AGE as covariate</span></span></span>
<span class="r-in"><span><span class="va">model.str</span> <span class="op">&lt;-</span> <span class="st">"~ 0 + SUBTYPE + AGE"</span></span></span>
<span class="r-in"><span><span class="va">coef.str</span> <span class="op">&lt;-</span> <span class="st">"SUBTYPE"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create contrasts</span></span></span>
<span class="r-in"><span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="paircomp.html">paircomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">oca.set</span><span class="op">)</span><span class="op">[</span>, <span class="va">coef.str</span><span class="op">]</span>, name <span class="op">=</span> <span class="va">coef.str</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Moderated t-statistics</span></span></span>
<span class="r-in"><span><span class="va">res1</span> <span class="op">&lt;-</span> <span class="fu">limma_contrasts</span><span class="op">(</span><span class="va">oca.set</span>, model.str <span class="op">=</span> <span class="va">model.str</span>,</span></span>
<span class="r-in"><span>                        coef.str <span class="op">=</span> <span class="va">coef.str</span>,</span></span>
<span class="r-in"><span>                        contrasts <span class="op">=</span> <span class="va">contrasts</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Moderated F-statistics</span></span></span>
<span class="r-in"><span><span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu">limma_contrasts</span><span class="op">(</span><span class="va">oca.set</span>, model.str <span class="op">=</span> <span class="va">model.str</span>,</span></span>
<span class="r-in"><span>                        coef.str <span class="op">=</span> <span class="va">coef.str</span>,</span></span>
<span class="r-in"><span>                        contrasts <span class="op">=</span> <span class="va">contrasts</span>, stat <span class="op">=</span> <span class="st">"F"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Count number of significant (&lt; 0.05) features</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">res2</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1059</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Robustify mean-variance trend (based on MDS plot)</span></span></span>
<span class="r-in"><span><span class="va">res3</span> <span class="op">&lt;-</span> <span class="fu">limma_contrasts</span><span class="op">(</span><span class="va">oca.set</span>, model.str <span class="op">=</span> <span class="va">model.str</span>,</span></span>
<span class="r-in"><span>                        coef.str <span class="op">=</span> <span class="va">coef.str</span>,</span></span>
<span class="r-in"><span>                        contrasts <span class="op">=</span> <span class="va">contrasts</span>,</span></span>
<span class="r-in"><span>                        trend <span class="op">=</span> <span class="cn">TRUE</span>, robust <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Count number of significant (&lt; 0.05) features in each contrast</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">res3</span><span class="op">$</span><span class="va">contrast</span>, <span class="va">res3</span><span class="op">$</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 FALSE TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Differentiated-Immunoreactive  8044   59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Mesenchymal-Immunoreactive     7904  199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Proliferative-Immunoreactive   7843  260</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

